src/Controls/src/SourceGen/RouteSyntaxReceiver.cs

```diff
@@ -0,0 +1,267 @@
+
+				var routeAttributes = cds.AttributeLists.SelectMany(al => al.Attributes)
+					.Where(a => a.Name.NormalizeWhitespace().ToFullString().Replace("Attribute", string.Empty) == "Route");
+
+				var routeAttribute = routeAttributes.FirstOrDefault();
+
+				var serviceAttributes = cds.AttributeLists.SelectMany(al => al.Attributes)
+					.Where(a => a.Name.NormalizeWhitespace().ToFullString().Replace("Attribute", string.Empty) == "MauiService");
+
+				var serviceAttribute = serviceAttributes.FirstOrDefault();
+
+				string lifetime;
+
+				if (routeAttribute is not null)
+				{
+					string? viewModelType = null;
+					var implicitViewModel = false;
+					var route = string.Empty;
+					var routes = new List<string>();
+
+					var routeArgument = true;
+					lifetime = Singleton;
+					arguments = routeAttribute.ArgumentList?.Arguments;
+
+					if (arguments is not null)
+					{
+						foreach (var argument in arguments)
+						{
+							if (argument.NameColon is not null)
+							{
+								switch (argument.NameColon.Name.Identifier.Text)
+								{
+									case "route":
+										route = GetRoute(cds, argument.Expression);
+										break;
+									case "lifetime":
+										lifetime = GetLifetime(argument.Expression);
+										break;
+								}
+							}
+							else if (argument.NameEquals is not null)
+							{
+								switch (argument.NameEquals.Name.Identifier.Text)
+								{
+									case "Routes" when argument.Expression is ArrayCreationExpressionSyntax arrayExpression:
+										if (arrayExpression?.Initializer?.Kind() == SyntaxKind.ArrayInitializerExpression)
+										{
+											routes.AddRange(arrayExpression.Initializer.Expressions.Select(expr => GetRoute(cds, expr)));
+										}
+										break;
+									case "Routes" when argument.Expression is ImplicitArrayCreationExpressionSyntax implicitArrayExpression:
+										if (implicitArrayExpression.Initializer.Kind() == SyntaxKind.ArrayInitializerExpression)
+										{
+											routes.AddRange(implicitArrayExpression.Initializer.Expressions.Select(expr => GetRoute(cds, expr)));
+										}
+										break;
+									case "Lifetime":
+										lifetime = GetLifetime(argument.Expression);
+										break;
+									case "ImplicitViewModel" when argument.Expression is LiteralExpressionSyntax literalExpression:
+										_ = bool.TryParse(literalExpression.Token.ValueText, out implicitViewModel);
+										break;
+									case "ViewModelType" when argument.Expression is TypeOfExpressionSyntax typeOfExpression:
+										viewModelType = ((IdentifierNameSyntax)typeOfExpression.Type).Identifier.Text;
+										break;
+								}
+							}
+							else if (argument.Expression is not null)
+							{
+								if (routeArgument)
+								{
+									routeArgument = false;
+									route = GetRoute(cds, argument.Expression);
+								}
+								else
+								{
+									lifetime = GetLifetime(argument.Expression);
+								}
+							}
```

Comment:
Probably can be re-written as:
```suggestion
							if (argument.Expression is not && and routeArgument)
							{
									routeArgument = false;
									route = GetRoute(cds, argument.Expression);
							}
							else
							{
								lifetime = GetLifetime(argument.Expression);
							}
```

Reply:
Both `route` and `lifetime` are to be evaluated only when the argument.Expression is not null as they're from the parameterized constructor and are based on the order of the arg definition.

Reply:
hmm do we need the else if as with others (due to being mutually exclusive)? Or can we remove the else, keep the nested if